extensions/example-extension/renderer.tsx

```diff
@@ -9,13 +9,16 @@ export default class ExampleExtension extends LensRendererExtension {
       title: "Example Extension",
       components: {
         Page: () => <ExamplePage extension={this}/>,
-      }
+      },
+      params: [
+        exampleNameUrlParam,
+      ]
```

Comment:
Isn't this specifying something the page already knows?

Reply:
No, it doesn't. It's locally defined url-param and it's passed here only you use default-values for this page (from menus for example, otherwise not required)

Reply:
Isn't this the same as not setting the `params` field at all?

Reply:
Yes, if they are don't have default values.

Reply:
Like in here:

```ts
export const exampleNameUrlParam = Navigation.createUrlParam<string>({
  name: "name",
  defaultValue: "demo",
});
```

Reply:
**UPD:** simplified default params definition for page-registration: 
```ts
params: {
  [name: string]: UrlParam | string /*default value*/ ;
}
``` 
So, if param is simple plain string then it's not required to use `createUrlParam` at all.

Reply:
Can you call them `createPageParam` and `PageParam` (Page instead of Url)? Lens knows internally that the param is part of the Url but the extension developer doesn't (or doesn't need) to know it. Internally we can keep Url but whatever is exposed could be more generic.

Reply:
Good idea, will update 👍